TowersOfHanoi/TowersOfHanoi.cs

```diff
@@ -1,12 +1,213 @@
 ﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace TowersOfHanoi
 {
     class Program
     {
-        static void Main(string[] args)
+        static void Main()
         {
-            Console.WriteLine("Hello World!");
+            //instatiates Game class
+            Game game = new Game();
+            //calls run method
+            game.run();
+        }
+    }
+    class Game
+    {
+        //creates Dictionary of towers with tower name as Key and Tower class as value
+        Dictionary<string, Tower> towers = new Dictionary<string, Tower>();
+
+        //method that runs game logic
+        public void run()
+        {
+            //while loop that runs only while game is not won
+            while(!CheckForWin())
+            {
+                //calls method that prints the game board
+                PrintBoard();
+
+                //asks user which tower to move block from then stores input as string
+                Console.WriteLine("Move block from Tower _? (A,B,C)");
+                string popOff = Console.ReadLine().ToUpper();
+
+                //asks user where to move block from selected tower to then stores input as string
+                Console.WriteLine("Move fromm tower "+popOff+" to tower_?");
+                string pushOn = Console.ReadLine().ToUpper();
+
+                //checks if move is legal
+                bool legal = IsLegal(popOff, pushOn);
+                if(legal)
+                {
+                    MovePiece(popOff, pushOn);
+                }
+                else
+                {
+                    Console.WriteLine("invalid move");
+                }
+                //seperates previous move from current move
+                Console.Clear();
+            }
+            //prints final board
+            PrintBoard();
+            //displays winning message
+            Console.WriteLine("Congratulations! You win!");
+        }
+        public Game()
+        { 
+            //instantiates 4 blocks
+            Block block1 = new Block(1,"   (=)");
```

Comment:
How would this be different if the number of blocks was passed in to the game class